pkg/apis/monitoring/v1alpha1/scrapeconfig_types.go

```diff
@@ -481,13 +481,24 @@ type ConsulSDConfig struct {
 	TLSConfig *v1.SafeTLSConfig `json:"tlsConfig,omitempty"`
 }
 
+type DNSRecordType string
```

Comment:
nit: all kubebuilder validation for struct?

Reply:
Not sure to follow here, could you elaborate?

Reply:
IIUC @dongjiang1989 means to move the `+kubebuilder:validation:Enum=A;AAAA;MX;NS;SRV` marker from L512 to here.

Reply:
Got it, that's a better idea.